File: Java/백준/Silver/2828. 사과 담기 게임/사과 담기 게임.java
//package algorithm.week2.q2828;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int t = scanner.nextInt();
        int l = 1;
        int r = l+m-1;
        int cnt = 0;
        for (int i = 0; i < t; i++) {
            int next = scanner.nextInt();
            int gap = 0;
            if(next>r){
                gap = next-r;
                cnt += gap;
                r+=gap;
                l+=gap;
                l = Math.min(l,n);
                r = Math.min(r,n);
            }else if(next<l){
                gap = l-next;
                cnt += gap;
                r-=gap;
                l-=gap;
                l = Math.max(l,1);
                r = Math.max(r,1);
            }
        }
        System.out.println(cnt);

    }
}

File: C++17/백준/Silver/2828. 사과 담기 게임/사과 담기 게임.cc
#include<bits/stdc++.h>
using namespace std;
int n,m,k,j,cnt;
	

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(0);cout.tie(0);
	cin>>n>>m;
	cin>>j;
	int l = 1;
	int r = l+m-1;
	
	for(int i=0;i<j;i++){
		int next;
		cin>>next;
		if(next>r){
			int gap = next-r;
			cnt+= gap;
			r += gap;
			l += gap;
			r = min(r,n);
			l = min(l,n);
		}else if(next<l){
			int gap =l-next; 
			cnt+= gap;
			l-= gap;
			r-= gap;
			l = max(1,l);
			r = max(1,r);
		}
//		cout<<"l : "<<l<<",r : "<<r<<"cnt: "<<cnt<<'\n';
	}
	cout<<cnt;
}
	
	
